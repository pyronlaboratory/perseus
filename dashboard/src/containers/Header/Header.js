import React from 'react';
import PropTypes from 'prop-types';

// material imports
import { withStyles } from '@material-ui/core/styles';

import AppBar from '@material-ui/core/AppBar';
import Grid from '@material-ui/core/Grid';
import Hidden from '@material-ui/core/Hidden';
import IconButton from '@material-ui/core/IconButton';
import Link from '@material-ui/core/Link';
import Toolbar from '@material-ui/core/Toolbar';
import Tooltip from '@material-ui/core/Tooltip';

// material icons
import GitHubIcon from '@material-ui/icons/GitHub';
import MenuIcon from '@material-ui/icons/Menu';

const lightColor = 'rgba(255, 255, 255, 0.7)';

const styles = (theme) => ({
  menuButton: {
    marginLeft: -theme.spacing(1),
  },
  link: {
    textDecoration: 'none',
    color: lightColor,
    '&:hover': {
      color: theme.palette.common.white,
    },
  }
});

function Header(props) {
  const { classes, onDrawerToggle } = props;

  return (
      <AppBar  style={{ backgroundColor: "#080a0e"}} position="sticky" elevation={0}>
        <Toolbar>
          <Grid container spacing={1} style={{alignItems:"center"}}>
            <Hidden smUp>
              <Grid item>
                <IconButton
                  color="inherit"
                  aria-label="open drawer"
                  onClick={onDrawerToggle}
                  className={classes.menuButton}
                >
                  <MenuIcon />
                </IconButton>
              </Grid>
            </Hidden>
            <Grid item xs />
            <Grid item>
              <Link className={classes.link} target="_blank" href="https://github.com/pyronlaboratory/perseus.git" variant="body2">                
                <Tooltip title="View Project Repository">
                  <IconButton color="inherit">
                    <GitHubIcon />
                  </IconButton>
                </Tooltip>
              </Link>
            </Grid>
          </Grid>
        </Toolbar>
      </AppBar>
  );
}

Header.propTypes = {
  classes: PropTypes.object.isRequired,
  onDrawerToggle: PropTypes.func.isRequired,
};

export default withStyles(styles)(Header);