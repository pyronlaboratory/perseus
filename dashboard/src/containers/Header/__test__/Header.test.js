import React from 'react';
import ReactDOM from 'react-dom';

// react testing
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import Header from './../Header';

afterEach(cleanup);

describe('header container', () => {
    it("renders without crashing", () => {
        const div = document.createElement("div");
        const onDrawerToggleMock = jest.fn();
        ReactDOM.render(
            <Header onDrawerToggle={onDrawerToggleMock} />
        , div);
    })

    it("matches snapshot", () => {
        const onDrawerToggleMock = jest.fn()
        const tree = renderer.create(<Header onDrawerToggle={onDrawerToggleMock} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})