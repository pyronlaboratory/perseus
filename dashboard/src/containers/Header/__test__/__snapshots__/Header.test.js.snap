// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header container matches snapshot 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation0"
  style={
    Object {
      "backgroundColor": "#080a0e",
    }
  }
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      />
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover Header-link-2 MuiTypography-body2 MuiTypography-colorPrimary"
          href="https://github.com/pyronlaboratory/perseus.git"
          onBlur={[Function]}
          onFocus={[Function]}
          target="_blank"
        >
          <button
            aria-describedby={null}
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            title="View Project Repository"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
  </div>
</header>
`;
